jetson/science_bridge/src/__main__.py

```diff
@@ -0,0 +1,391 @@
+            self.ser.write(bytes(message, encoding='utf-8'))
+        print("Mosfet Received")
+        pass
+
+    def nav_status(self, channel, msg):  # TODO - fix make better
+        print("Received nav req")
+        # Off, Done, Else
+
+        struct = NavStatus.decode(msg)
+        message = "$Mosfet,{device},{enable},11111111"  # TODO - PROBABLY OUTDATED, WRONG PADDING
+        # All Leds are 1 digit so hardcode in padding
+
+        # Off = Blue
+        if struct.nav_state_name == "Off":
+            print("navstatus off")
+            offmessage = message.format(device=2, enable=1) + "1"
+            self.ser.write(bytes(offmessage, encoding='utf-8'))
+            prev = 2
+        # Done = Flashing green
+        elif struct.nav_state_name == "Done":
+            print("navstatus Done")
+            # Flashing by turning on and off for 1 second intervals
+            # Maybe change to
+            for i in range(0, 6):  # TODO - CHANGE TO RANGE (6)
+                self.ser.write(bytes(message.format(device=1, enable=1), encoding='utf-8'))
+                time.sleep(1)
+                self.ser.write(bytes(message.format(device=1, enable=0), encoding='utf-8'))
+                time.sleep(1)
+                prev = 1
+        # Everytime else = Red
+        else:
+            print("navstatus else")
+            messageon = message.format(device=0, enable=1)
+            self.ser.write(bytes(messageon, encoding='utf-8'))
+            prev = 0
+        time.sleep(1)
+        # Green should be in a finished state so no need to turn it off
+        if (prev != 2):
+            self.ser.write(bytes(message.format(device=2, enable=0),
+                                 encoding='utf-8'))
+        if (prev != 0):
+            self.ser.write(bytes(message.format(device=0, enable=0),
+                                 encoding='utf-8'))
+
+    def servo_transmit(self, channel, msg):
+        # get cmd lcm and send to nucleo
+        struct = ServoCmd.decode(msg)
+        print("Received Servo Cmd")
+        # parse data into expected format
+        message = "$Servo,{angle0},{angle1},{angle2}"
+        message = message.format(angle0=struct.angle0, angle1=struct.angle1, angle2=struct.angle2)
+        print(message)
+        while(len(message) < 30):
+            message += ","
+        self.ser.close()
+        self.ser.open()
+        if self.ser.isOpen():
+            self.ser.write(bytes(message, encoding='utf-8'))
+
+    def carousel_openloop_transmit(self, channel, msg):
+        struct = CarouselOpenLoopCmd.decode(msg)
+
+        if (self.last_openloop_cmd == struct.throttle):
+            return
+        print("Received Carousel (OL) Cmd")
+        self.last_openloop_cmd = struct.throttle
+        # parse data into expected format
+        message = "$OpenCarousel,{throttle}"
+        message = message.format(throttle=struct.throttle)
+        print(message)
+        while(len(message) < 30):
+            message += ","
```

Comment:
why is the padding here commas??? can this add padding become a function that takes in the msg & returns it padded? this is like the 6th time i've seen it